src/runInContext.js

```diff
@@ -0,0 +1,235 @@
+              stringify(null, value) === "1" &&
+              stringify([1, 2], null, 1) == "[\n 1,\n 2\n]"
+            );
+          });
+        }
+        has[name] = isSupported;
+        return isSupported;
+      }
+      if (name == "json-parse") {
+        // Test `JSON.parse`.
+        var parse = parse, isSupported = typeof parse == "function";
+        if (isSupported) {
+          isSupported = !!attempt(function () {
+            // FF 3.1b1, b2 will throw an exception if a bare literal is provided.
+            // Conforming implementations should also coerce the initial argument to
+            // a string prior to parsing.
+            if (parse("0") !== 0 || parse(false)) {
+              return;
+            }
+            // Simple parsing test.
+            var value = parse(serialized);
+            return value["a"].length == 5 && value["a"][0] === 1;
+          });
+          if (isSupported) {
+            isSupported = !!attempt(function () {
+              // Safari <= 5.1.2 and FF 3.1b1 allow unescaped tabs in strings.
+              return !parse('"\t"');
+            });
+          }
+          if (isSupported) {
+            isSupported = !!attempt(function () {
+              // FF 4.0 and 4.0.1 allow leading `+` signs and leading
+              // decimal points. FF 4.0, 4.0.1, and IE 9-10 also allow
+              // certain octal literals.
+              return parse("01") !== 1;
+            });
+          }
+          if (isSupported) {
+            isSupported = !!attempt(function () {
+              // FF 4.0, 4.0.1, and Rhino 1.7R3-R4 allow trailing decimal
+              // points. These environments, along with FF 3.1b1 and 2,
+              // also allow trailing commas in JSON objects and arrays.
+              return parse("1.") !== 1;
+            });
+          }
+        }
+        has[name] = isSupported;
+        return isSupported;
+      }
+      return false;
+    }
+    has["bug-string-char-index"] = null;
+    has["extended-years"] = null;
+    has["json-stringify"] = null;
+    has["date-serialization"] = null; // Depends on `json-stringify`, `extended-years`.
+    has["json-parse"] = null; // Depends on `json-stringify`, `json-parse`, `date-serialization`.
+
+    if (has("json")) {
+      return;
+    }
+
+    var forOwn = makeForOwn(getClass, Object.prototype.hasOwnProperty);
+
+    if (!has("json-stringify")) {
+      exports.stringify = makeStringify(getClass, forOwn, TypeError);
+    } else if (!has("date-serialization")) {
+      // For environments with `JSON.stringify` but buggy date serialization,
+      // we override the native `Date#toJSON` implementation with a
+      // spec-compliant one.
+      var UTCDate;
+      if (!has("extended-years")) {
+        UTCDate = makeUTCDate(Math.floor);
+      }
+      var serializeDate = makeSerializeDate(UTCDate);
+      if (has("json-stringify")) {
+        // TODO: Revert via `JSON3.noConflict()`.
```

Comment:
Making this work with `noConflict` is going to be interesting, since `noConflict` isn't a module, but a browser-only function that's injected by the webpack `WrapperPlugin`. We'd need to make `noConflict` context-aware...